server/src/com/thoughtworks/go/server/service/GoConfigService.java

```diff
@@ -148,12 +148,13 @@ private boolean canEditPipeline(String pipelineName, Username username, Localize
     }
 
     public boolean canEditPipeline(String pipelineName, Username username) {
-        if (pipelineExists(pipelineName) &&
-                isUserAdminOfGroup(username.getUsername(), findGroupNameByPipeline(new CaseInsensitiveString(pipelineName)))) {
-            return true;
+        PipelineConfig pipelineConfig;
+        try {
+            pipelineConfig = pipelineConfigNamed(new CaseInsensitiveString(pipelineName));
```

Comment:
instead of duplicating the logic of `pipelineExists` over here, can you not check for `pipelineExists(pipelineName) && pipelineConfigNamed(new CaseInsensitiveString(pipelineName)).getOrigin().canEdit()` ?